packages/next-auth/src/providers/email.ts

```diff
@@ -14,10 +21,9 @@ export interface SendVerificationRequestParams {
   theme: Theme
 }
 
-export interface EmailConfig extends CommonProviderOptions {
-  type: "email"
-  // TODO: Make use of https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html
-  server: string | SMTPTransportOptions
+export interface EmailUserConfig {
+  server: AllTransportOptions
```

Comment:
```suggestion
  server?: AllTransportOptions
```

Actually, even this can be optional, for example when using an HTTP based Email service: https://authjs.dev/guides/providers/email-http